hack/install.sh

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env bash
+set -e
+
+DEFAULT_BIN_DIR="/usr/local/bin"
+BIN_DIR=${1:-"${DEFAULT_BIN_DIR}"}
+GITHUB_REPO="karmada-io/karmada"
+
+# Helper functions for logs
+info() {
+    echo '[INFO] ' "$@"
+}
+
+warn() {
+    echo '[WARN] ' "$@" >&2
+}
+
+fatal() {
+    echo '[ERROR] ' "$@" >&2
+    exit 1
+}
+
+# Set os, fatal if operating system not supported
+setup_verify_os() {
+    if [[ -z "${OS}" ]]; then
+        OS=$(uname)
+    fi
+    case ${OS} in
+        Darwin)
+            OS=darwin
+            ;;
+        Linux)
+            OS=linux
+            ;;
+        *)
+            fatal "Unsupported operating system ${OS}"
+    esac
+}
+
+# Set arch, fatal if architecture not supported
+setup_verify_arch() {
+    if [[ -z "${ARCH}" ]]; then
+        ARCH=$(uname -m)
+    fi
+    case ${ARCH} in
+        arm64|aarch64|armv8l)
+            ARCH=arm64
+            ;;
+        amd64)
+            ARCH=amd64
+            ;;
+        x86_64)
+            ARCH=amd64
+            ;;
+        *)
+            fatal "Unsupported architecture ${ARCH}"
+    esac
+}
+
+# Set cli tools, install karmadactl by default, fatal if cli tool not supported
+setup_install_cli() {
+    if [[ -z "${CLI}" ]]; then
+        CLI=karmadactl
+        return
+    fi
+    case ${CLI} in
+        karmadactl)
+          CLI=karmadactl
+          ;;
+        kubectl-karmada)
+          CLI=kubectl-karmada
+          ;;
+        *)
+            fatal "Unsupported cli tool ${CLI}"
+    esac
+}
+
+# Verify existence of downloader executable
+verify_downloader() {
+    # Return failure if it doesn't exist or is no executable
+    [[ -x "$(which "$1")" ]] || return 1
+
+    # Set verified executable as our downloader program and return success
+    DOWNLOADER=$1
+    return 0
+}
+
+# Create tempory directory and cleanup when done
+setup_tmp() {
+    TMP_DIR=$(mktemp -d -t "${CLI}"-install.XXXXXXXXXX)
+    TMP_METADATA="${TMP_DIR}/${CLI}.json"
+    TMP_HASH="${TMP_DIR}/${CLI}.hash"
+    TMP_BIN="${TMP_DIR}/${CLI}.tgz"
+    cleanup() {
+        local code=$?
+        set +e
+        trap - EXIT
+        rm -rf "${TMP_DIR}"
+        exit ${code}
+    }
+    trap cleanup INT EXIT
+}
+
+# Find version from Github metadata
+get_release_version() {
+    if [[ -n "${KARMADA_VERSION}" ]]; then
+      SUFFIX_URL="tags/v${KARMADA_VERSION}"
+    else
+      SUFFIX_URL="latest"
+    fi
+
+    METADATA_URL="https://api.github.com/repos/${GITHUB_REPO}/releases/${SUFFIX_URL}"
+
+    info "Downloading metadata ${METADATA_URL}"
+    download "${TMP_METADATA}" "${METADATA_URL}"
+
+    VERSION_KARMADA=$(grep '"tag_name":' "${TMP_METADATA}" | sed -E 's/.*"([^"]+)".*/\1/' | cut -c 2-)
+    if [[ -n "${VERSION_KARMADA}" ]]; then
+        info "Using ${VERSION_KARMADA} as release"
+    else
+        fatal "Unable to determine release version"
+    fi
+
+    set +e
+    vercomp ${VERSION_KARMADA} 1.2.0
```

Comment:
Only support above 1.2.0 because these have checksum?
Could we support below version without checksum verify?

Reply:
There is one of a reason. Another is that the release before 1.2.0, we only output the kubectl-karmada file, which means the scripts cannot install karmdactl before v1.2.0.

Reply:
But users expect to install v1.2.0  kubectl-karmada with
```
export CLI=kubectl-karmada
export KARMADA_VERSION=kubectl-karmada
./install.sh 
```

Reply:
Yes, we can offer the installation of kubectl-karmada before v1.2.0. I guess it's not security enough to use. cc @RainbowMango for opinions.

Reply:
@RainbowMango what do you think about it?

Reply:
I'm ok that this script just supports v1.2.0+. For versions before v1.2, seems we just have `kubectl-karmada` in releases([like v1.1.0](https://github.com/karmada-io/karmada/releases/tag/v1.1.0)).